lib/spack/spack/ci.py

```diff
@@ -14,31 +18,36 @@
 
 import llnl.util.tty as tty
 
-import spack.environment as ev
+import spack
+import spack.binary_distribution as bindist
+import spack.cmd.buildcache as buildcache
 import spack.compilers as compilers
+import spack.config as cfg
+import spack.environment as ev
 from spack.dependency import all_deptypes
 from spack.error import SpackError
 import spack.hash_types as ht
+from spack.main import SpackCommand
+import spack.repo
 from spack.spec import Spec
 import spack.util.spack_yaml as syaml
+import spack.util.web as web_util
 
-description = "generate release build set as .gitlab-ci.yml"
-section = "build"
-level = "long"
 
+spack_gpg = SpackCommand('gpg')
+spack_compiler = SpackCommand('compiler')
 
-def setup_parser(subparser):
-    subparser.add_argument(
-        '-o', '--output-file', default=".gitlab-ci.yml",
-        help="path to output file to write")
 
-    subparser.add_argument(
-        '-p', '--print-summary', action='store_true', default=False,
-        help="Print summary of staged jobs to standard output")
+class TemporaryDirectory(object):
+    def __init__(self):
+        self.temporary_directory = tempfile.mkdtemp()
 
-    subparser.add_argument(
-        '--cdash-credentials', default=None,
-        help="Path to file containing CDash authentication token")
+    def __enter__(self):
+        return self.temporary_directory
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        shutil.rmtree(self.temporary_directory)
+        return False
 
 
 def _create_buildgroup(opener, headers, url, project, group_name, group_type):
```

Comment:
there should probably eventually be a `spack.util.cdash` module.  Doesn't have to happen now but don't be afraid to make core modules :)